compiler/src/iree/compiler/Dialect/Flow/Transforms/ConvertLinalgMatmulToMmt4D.cpp

```diff
@@ -331,45 +332,58 @@ LinalgMatmulOpToLinalgMmt4DOpPattern::chooseTileParams(Value lhs, Value rhs,
   int64_t shapeN = rhsType.getShape()[1];
   auto chooseMatMulOrMatVec = [=](ArrayRef<int> m0k0n0,
                                   ArrayRef<int> m0k0n0ForMatVec,
+                                  ArrayRef<int> m0k0n0ForWhenRhsHas2Columns,
                                   std::string comment) {
     assert(m0k0n0ForMatVec[2] == 1 && "not a matrix*vector shape");
-    if (shapeN == 1) {
-      return Mmt4DTileParams(m0k0n0ForMatVec, comment + ", matrix*vector");
-    } else if (shapeM == 1) {
+    assert(m0k0n0ForWhenRhsHas2Columns[2] == 2 && "not a matrix*vector2 shape");
+
+    SmallVector<int> config;
```

Comment:
`params` would be a better name than `config` (try not to introduce a new term).